File: node/packages/fluent-cards/src/builders/AuthenticationBuilder.ts
import type { AuthenticationConfiguration, AuthCardButton, TokenExchangeResource } from '../models.js';

/** Fluent builder for {@link AuthenticationConfiguration}. */
export class AuthenticationBuilder {
  private readonly auth: AuthenticationConfiguration = {};

  withText(text: string): this { this.auth.text = text; return this; }
  withConnectionName(connectionName: string): this { this.auth.connectionName = connectionName; return this; }

  withTokenExchangeResource(resource: TokenExchangeResource): this {
    this.auth.tokenExchangeResource = resource;
    return this;
  }

  addButton(button: AuthCardButton): this {
    (this.auth.buttons ??= []).push(button);
    return this;
  }

  build(): AuthenticationConfiguration { return this.auth; }
}

File: python/src/fluent_cards/builders/authentication_builder.py
from __future__ import annotations


class AuthenticationBuilder:
    def __init__(self):
        self._auth: dict = {}

    def with_text(self, text: str) -> AuthenticationBuilder:
        self._auth['text'] = text
        return self

    def with_connection_name(self, connection_name: str) -> AuthenticationBuilder:
        self._auth['connectionName'] = connection_name
        return self

    def with_token_exchange_resource(self, resource: dict) -> AuthenticationBuilder:
        self._auth['tokenExchangeResource'] = resource
        return self

    def add_button(self, button: dict) -> AuthenticationBuilder:
        if 'buttons' not in self._auth:
            self._auth['buttons'] = []
        self._auth['buttons'].append(button)
        return self

    def build(self) -> dict:
        return self._auth
